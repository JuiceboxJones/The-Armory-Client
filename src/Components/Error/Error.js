import React from 'react';

export default function Error(props) {
  return (
    <div className="error-container">
      <p>{props.error}</p>
      {/* icon here */}
      <button type="button" onClick={props.close}>X</button>
    </div>
  )
}