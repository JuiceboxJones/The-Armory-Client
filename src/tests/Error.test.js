import React from 'react';
import Error from '../Components/Error/Error';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('FAQ component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Error />
      </MemoryRouter>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  })

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Error />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})