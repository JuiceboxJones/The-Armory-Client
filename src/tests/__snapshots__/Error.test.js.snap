// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ component renders the UI as expected 1`] = `
<div
  className="error-container"
>
  <p />
  <button
    type="button"
  >
    X
  </button>
</div>
`;
